src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -80,50 +80,57 @@ static constexpr HexVertexIndex edgeAdjacentFacet[12][2]{
 };
 
 
-// Structure for face computation
-// TODO Optimize: replace (v0, v1, v2, f) by (v0*nbNodes + v1, v2, f)
+/* Structure for face computation
+ * The table stores 3 sorted LocalVertexIndex v0 < v1 < v2
+ * and f the LocalFaceIndex that identifies unambiguously a face in a cell
+ *  v3 =  Hex::nbFacets * idCell + f_hex
+ * where f_hex is the HexFaceIndex
+ * 
+ * TODO Optimize: replace (v0, v1, v2) by (v0*nbNodes + v1, v2)
+ */
 struct FaceInfo{
-  localIndex v[4];
-};
 
-struct {
-  bool operator()( FaceInfo const & a, FaceInfo const & b) const {
-    if(a.v[0] != b.v[0]) return a.v[0] < b.v[0];
-    if(a.v[1] != b.v[1]) return a.v[1] < b.v[1];
-    if(a.v[2] != b.v[2]) return a.v[2] < b.v[2];
+  /* Compare the indices of the 3 sorted vertex indices
+  */
+  bool operator<( FaceInfo const & right) const 
+  {
+    if(v[0] != right.v[0]) return v[0] < right.v[0];
+    if(v[1] != right.v[1]) return v[1] < right.v[1];
+    if(v[2] != right.v[2]) return v[2] < right.v[2];
     return false;
   }
-} compareFaceInfo;
 
-// Structure for edge computation and storage
+  LocalVertexIndex v[3];
+  LocalFaceIndex cellFace;
+};
+
+
+/* Structure for edge computation and storage
+ */
 struct EdgeInfo
 {
   EdgeInfo(LocalVertexIndex a, LocalEdgeIndex b){
    first = a;
    second = b;
   }
-  EdgeInfo(){
-   first = 0;
-   second = 0;
-  }
-  LocalVertexIndex first;
-  LocalEdgeIndex second;
-};
+  EdgeInfo(): EdgeInfo( 0, 0 ) {}
 
-struct {
-  bool operator() ( EdgeInfo const & a, EdgeInfo const & b ) const
+  bool operator==( EdgeInfo const & right ) const 
```

Comment:
Same here. Having two `EdgeInfo` considered equal even if their second node is different could be a bit hidden and reserved to the specific function where you use them I think.

Reply:
I agree to disagree. This isinherent and tightly linked to the way the EgdeInfo is designed. Two edges are the same is they have the same nodes, ie if the first values match.